sklearn/utils/metaestimators.py

```diff
@@ -10,11 +10,30 @@
 __all__ = ['if_delegate_has_method']
 
 
+def _hasattr_nested(obj, attr):
```

Comment:
@amueller I haven't tried implement this using `try except attrgetter` because I'm worried it may affect performance. Do you think it would be better than the current approach?
```Python
    def __get__(self, obj, type=None):
        # raise an AttributeError if the attribute is not present on the object
        if obj is not None:
            # delegate only on instances, not the classes.
            # this is to allow access to the docstrings.
            ecpt = None
            for item in self.attrs_tuple:
                try:
                    attrgetter(item)(obj)
                    break
                except ValueError as e:
                    ecpt = e
            else:
                raise(ecpt)
```